### PR TITLE
Load MySQL variables from environment

### PR DESCRIPTION
**Pull request by ranisalt**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Loads MySQL configuration from the environment first, keeping compatibility with Lua config.

This allows better compatibility with database containers, such as [Docker's mariadb](https://hub.docker.com/_/mariadb) and a single source of configuration can be used for the database. It's also easier for other software, such as AACs, that can load the credentials from the environment rather than poorly parsing the Lua config.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/configmanager.cpp [hunk 1]
```diff
@@ -170,10 +178,10 @@ bool ConfigManager::load()
...
-		string[MYSQL_USER] = getGlobalString(L, "mysqlUser", "forgottenserver");
-		string[MYSQL_PASS] = getGlobalString(L, "mysqlPass", "");
-		string[MYSQL_DB] = getGlobalString(L, "mysqlDatabase", "forgottenserver");
+		string[MYSQL_HOST] = getGlobalString(L, "mysqlHost", getEnv("MYSQL_HOST", "127.0.0.1"));
+		string[MYSQL_USER] = getGlobalString(L, "mysqlUser", getEnv("MYSQL_USER", "forgottenserver"));
+		string[MYSQL_PASS] = getGlobalString(L, "mysqlPass", getEnv("MYSQL_PASSWORD", ""));
+		string[MYSQL_DB] = getGlobalString(L, "mysqlDatabase", getEnv("MYSQL_DATABASE", "forgottenserver"));
 		string[MYSQL_SOCK] = getGlobalString(L, "mysqlSock", "");
 
 		integer[SQL_PORT] = getGlobalNumber(L, "mysqlPort", 3306);
```
Comment (DSpeichert) [inline src/configmanager.cpp:187]: Why not extend environment support to socket and port?

Comment (ranisalt) [reply src/configmanager.cpp:187]: Unintentional, will fix

**Review by DSpeichert — APPROVED** [review]

**Review by moviebr — APPROVED** [review]
Nice

**Review by EPuncker — APPROVED** [review]

**Comment by slawkens** [comment]
Would be nice to have an .env.example file, so there is something just to fill out and the user doesn't need to create it himself.

**Comment by gesior** [comment]
> This allows better compatibility with database containers, such as [Docker's mariadb](https://hub.docker.com/_/mariadb) and a single source of configuration can be used for the database

In most (all?) other config/env loaders in PHP/Python/JS/TS it works different:
Env variables overwrite config file variables.
In current implementation, it loads env variables only, if there is no SQL config in `config.lua` - which results in removing SQL config from `config.lua`!

I think it should work different: it should load `config.lua` first, then check, if there are env variables defined (not empty?) and overwrite loaded `config.lua` values.

This PR changes TFS basic config and makes it pretty confusing for newbie users who use any tutorial 'how to run first OTS' (ex. https://otland.net/threads/cannot-open-config-lua-no-such-file-or-directory.290452/#post-2763504 ), as there is no MySQL config anymore in `config.lua`.

@ranisalt 
**What do you think about it?**
I can create PR that will make it work as I described above, if you think it's ok.

**Comment by ranisalt** [comment]
@gesior you're correct, just need swapping the order to call `getEnv` with `getGlobalString` as a fallback and not the other way around. Feel free to open it or I'll to it soon

**Pull request merged**